lib/assessment/label.py

```diff
@@ -334,21 +347,21 @@ def compute_messages(self, prompt, rubric, student_code, examples=[]):
         messages.append({'role': 'user', 'content': student_code})
         return messages
 
-    def get_response_data_if_valid(self, response_text, rubric, student_id, choice_index=None, reraise=False, response_type='tsv'):
-        choice_text = f"Choice {choice_index}: " if choice_index is not None else ''
-        if not response_text:
-            logging.error(f"{student_id} {choice_text} Invalid response: empty response")
-            return None
-        text = response_text.strip()
-
-        if response_type == 'json':
-            response_data = self.parse_json_response(text, student_id)
-        elif response_type == 'tsv':
-            response_data = self.parse_non_json_response(text)
-        else:
-            raise ValueError(f"Invalid response type: {response_type}")
-
+    def get_response_data_if_valid(self, choice, rubric, student_id, choice_index=None, reraise=False, response_type='tsv'):
+        response_text = choice['message']['content']
```

Comment:
refactor: pass entire choice in to this method, so we can access the finish_reason